packages/core/src/breadcrumb/breadcrumb.element.ts

```diff
@@ -0,0 +1,74 @@
+import { baseStyles, internalProperty } from '@cds/core/internal';
+import { html, LitElement } from 'lit-element';
+import { styles } from './breadcrumb.element.css.js';
+
+export const CdsBreadcrumbTagName = 'cds-breadcrumb';
+
+interface ItemEntry {
+  element: Element;
+  separator: Element;
+}
+
+/**
+ * ```typescript
+ * import '@cds/core/breadcrumb/register.js';
+ * ```
+ *
+ * ```html
+ * <cds-breadcrumb aria-label="breadcrumb">
+ *   <a href="/home" cds-text="link">Home</a>
+ *   <a href="/parent" cds-text="link">Parent page</a>
+ *   <span aria-current="page">Current page</span>
+ * </cds-breadcrumb>
+ * ```
+ *
+ * @beta
+ * @element cds-breadcrumb
+ * @cssprop --font-size
+ * @cssprop --font-weight
+ * @cssprop --letter-spacing
+ * @cssprop --color
+ */
+export class CdsBreadcrumb extends LitElement {
+  @internalProperty({ type: String, reflect: true, attribute: 'role' })
+  protected role = 'navigation';
+
+  render() {
+    return html`
+      <div class="private-host">
+        <ol cds-layout="horizontal gap:sm align:vertical-center">
+          ${this.getItems().map(
+            item =>
+              html`<li cds-layout="horizontal gap:sm align:vertical-center">
+                ${item.element} ${item.separator}
+              </li>`
+          )}
+        </ol>
+      </div>
+    `;
+  }
+
+  getItems(): ItemEntry[] {
+    const customSeparator = Array.from(this.children).find(c => c.getAttribute('slot') === 'cds-separator');
```

Comment:
@jeeyun : Do you think this work here could be simplified by using `@querySlot` or `@querySlotAll`?

Reply:
Yes, we should be using `@querySlot` here. Thank you for catching that.

I just realized that the code for rendering the items and separators has become more imperative. We try to make our templates to be as declarative as possible, to see more easily the markup that you'd see in the rendered version. Here are my proposed changes:

```
export class CdsBreadcrumb extends LitElement {
  ...
  // add the following (querySlot can be imported from '@cds/core/internal')
  @querySlot('[slot="cds-separator"]') private customSeparator: HTMLElement;

 render() {
    return html`
      <div class="private-host">
        <ol cds-layout="horizontal gap:sm align:vertical-center">
          ${this.getItems().map(
            item => html`
              <li cds-layout="horizontal gap:sm align:vertical-center">
                <slot name="${item.getAttribute("slot")}"></slot>
                <span part="separator" aria-hidden="true">
                  ${this.separator}
                </span>
              </li>
            `
          )}
        </ol>
      </div>
    `;
  }

  get separator() {
    let separator: Element;
    if (this.customSeparator) {
      separator = this.customSeparator.cloneNode(true) as Element;
      separator.removeAttribute('slot');
    } 
    return html`${separator? separator : html`/`}`;
  }

  getItems(): Element[] {
    return Array.from(this.children)
      .filter(c => c.getAttribute('slot') !== 'cds-separator')
      .map(element => {
        element.setAttribute("slot", createId());
        return element;
      });
  }
  ...
}
```